### ISSUE TITLE
ExJSON.parse(_, :to_map) places an empty Map inside empty arrays when parsed

### ISSUE THREAD

**Issue by adamkittelson** [guedes/exjson#26]
Parsing Arrays with and without contents works when parsing :to_keyword:

````
iex> ExJSON.parse("{\"numbers\": [1,2,3]}") 
[{"numbers", [1, 2, 3]}]

iex> ExJSON.parse("{\"numbers\": []}")         
[{"numbers", []}]
````

And parsing an array with contents works when parsing :to_map:
````
iex> ExJSON.parse("{\"numbers\": [1, 2, 3]}", :to_map) 
%{"numbers" => [1, 2, 3]}
````

But when parsing an empty array :to_map it puts an empty map in the array:
````
iex> ExJSON.parse("{\"numbers\": []}", :to_map)       
%{"numbers" => [%{}]}
````

**Comment by guedes** [comment]
Closed in #29 thank you for your time and work on this.

**Comment by timruffles** [comment]
Welcome! Thanks for the library :)
